src/operator/numpy/np_tensordot_op-inl.h

```diff
@@ -0,0 +1,391 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file np_tensordot_op-inl.h
+ * \brief CPU Implementation of numpy-compatible tensordot
+ */
+
+
+#include "np_matrix_op-inl.h"
+
+namespace mxnet {
+namespace op {
+
+using namespace mxnet;
+using namespace mshadow;
+
+struct TensordotParam : public dmlc::Parameter<TensordotParam> {
+  mxnet::Tuple<int> a_axes_summed, b_axes_summed; 
+  DMLC_DECLARE_PARAMETER(TensordotParam) {
+    DMLC_DECLARE_FIELD(a_axes_summed);
+    DMLC_DECLARE_FIELD(b_axes_summed);
+  }
+};
+
+/**
+ * Gets matrix dimensions of a and b after transpose and reshape. 
+ */
+inline void GetMatrixDimensions(
+    int& ad1, 
```

Comment:
Pass by pointers if you want to modify the argument. Please make sure the coding style can pass the lint check.

Reply:
Fixed. Thx.